File: frontend/src/types/album.ts
import type { PhotoItem } from '@/types/photo'

/** 相册（对齐后端 AlbumResponse） */
export interface Album {
  id: string
  owner_id: string
  name: string
  description?: string | null
  cover_photo_id?: string | null
  is_system: boolean
  album_type: string
  conditions?: Record<string, unknown> | null
  photo_count: number
  created_at: string
  updated_at?: string | null
}

/** 相册内照片列表响应 */
export interface AlbumPhotosResponse {
  total: number
  page: number
  page_size: number
  items: PhotoItem[]
}

/** 创建相册入参 */
export interface AlbumCreateParams {
  name: string
  description?: string
  album_type?: string
  conditions?: Record<string, unknown>
}

/** 更新相册入参 */
export interface AlbumUpdateParams {
  name?: string
  description?: string
  cover_photo_id?: string
}

File: backend/app/schemas/album.py
"""
相册相关 Schema
"""
import uuid
from typing import Optional, Any
from datetime import datetime
from pydantic import BaseModel, Field, field_validator, model_validator


class AlbumCreate(BaseModel):
    """创建相册"""
    name: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    album_type: str = "manual"
    conditions: Optional[dict] = None

    @model_validator(mode="after")
    def _check_smart_conditions(self) -> "AlbumCreate":
        if self.album_type in ("smart", "conditional") and not self.conditions:
            raise ValueError("智能相册（smart/conditional）必须提供 conditions 筛选规则")
        return self


class AlbumUpdate(BaseModel):
    """更新相册"""
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    cover_photo_id: Optional[str] = None


class AlbumResponse(BaseModel):
    """相册响应"""
    id: str
    owner_id: str
    name: str
    description: Optional[str] = None
    cover_photo_id: Optional[str] = None
    is_system: bool = False
    album_type: str = "manual"
    conditions: Optional[dict] = None
    photo_count: int = 0
    created_at: datetime
    updated_at: Optional[datetime] = None

    @field_validator("id", "owner_id", "cover_photo_id", mode="before")
    @classmethod
    def coerce_uuid(cls, v: Any) -> Optional[str]:
        if v is None:
            return None
        return str(v) if isinstance(v, uuid.UUID) else v

    model_config = {"from_attributes": True}
